dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureSet.java

```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package componenttest.rules.repeater;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * An immutable set of features with an ID
+ */
+public class FeatureSet {
+    private final String id;
+    private final Set<String> features;
+
+    /**
+     * Create a new FeatureSet with the given ID and set of features
+     *
+     * @param id       The ID of the FeatureSet. Must be unique.
+     * @param features The features to include in the set
+     */
+    public FeatureSet(String id, Set<String> features) {
+        if (id == null)
+            throw new NullPointerException();
+        this.id = id;
+        this.features = Collections.unmodifiableSet(new HashSet<>(features));
+    }
+
+    /**
+     * Get the ID of this FeatureSet
+     *
+     * @return the id
+     */
+    public final String getID() {
+        return this.id;
+    }
+
+    /**
+     * Get an unmodifiable set of the features
+     *
+     * @return the features
+     */
+    public final Set<String> getFeatures() {
+        return this.features;
+    }
+
+    /**
+     * Create a new FeatureSetBuilder based on this FeatureSet
+     *
+     * @param  feature the feature to add to the set
+     * @return         a FeatureSetBuilder
+     */
+    public FeatureSetBuilder addFeature(String feature) {
+        FeatureSetBuilder builder = new FeatureSetBuilder(this);
+        builder.addFeature(feature);
+        return builder;
+    }
+
+    /**
+     * Create a new FeatureSetBuilder based on this FeatureSet
+     *
+     * @param  feature the feature to remove from the set
+     * @return         a FeatureSetBuilder
+     */
+    public FeatureSetBuilder removeFeature(String feature) {
+        FeatureSetBuilder builder = new FeatureSetBuilder(this);
+        builder.removeFeature(feature);
+        return builder;
+    }
+
+    @Override
+    public int hashCode() {
+        return id.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+
+        FeatureSet other = (FeatureSet) obj;
+
+        if (id == null) {
+            if (other.id != null)
+                return false;
+        } else if (!id.equals(other.id))
+            return false;
+
+        return true;
+    }
```

Comment:
We should probably document what equality means for this class if we're overriding `equals` and `hashCode`

Reply:
done